app/models/user.rb

```diff
@@ -521,6 +521,10 @@ def reload(...)
     super(...)
   end
 
+  def last_sign_in_email_address
+    confirmed_email_addresses.first
+  end
+
```

Comment:
I like how you are using this to simplify things like `EmailContext.new(current_user)...etc`